internal/workflow/continue.go

```diff
@@ -333,6 +346,18 @@ func (w *Session) continueExec(msg *dipper.Message, exports []map[string]interfa
 		w.iteration++
 		w.executeIteration(msg)
 	case WorkflowNextParallelIteration:
+		if w.workflow.IteratePool != "" {
+			poolCount := dipper.Must(strconv.Atoi(w.workflow.IteratePool)).(int)
+			if poolCount > 0 {
+				w.iterationLock.Lock()
+				defer w.iterationLock.Unlock()
+				i := poolCount + int(w.iteration)
+				if i < w.lenOfIterate() {
```

Comment:
I'm a little confused what is going on here.

`poolCound + w.iteration < lenOfIterate`

?

Reply:
poolCount basically means running iterations, and the w.iteration is the finished one. So poolCount + w.iterations gives us the next iteration to start, if it is still lower than the total number of iterations.